src/app/src/main/kotlin/org/imperial/mrc/hint/emails/EmailManager.kt

```diff
@@ -0,0 +1,27 @@
+package org.imperial.mrc.hint.emails
+
+import org.imperial.mrc.hint.AppProperties
+import org.imperial.mrc.hint.security.tokens.OneTimeTokenManager
+
+interface EmailManager {
+    fun sendEmail(data: EmailData, emailAddress: String)
+    fun sendPasswordEmail(emailAddress: String, username: String, emailTemplate: PasswordEmailTemplate)
+}
+
+abstract class BaseEmailManager(private val appProperties: AppProperties,
+                                private val oneTimeTokenManager: OneTimeTokenManager) : EmailManager {
+
+    override fun sendPasswordEmail(emailAddress: String, username: String, emailTemplate: PasswordEmailTemplate) {
+        val token = oneTimeTokenManager.generateOnetimeSetPasswordToken(username)
```

Comment:
This also seems a little bit odd from a responsibilities perspective - for the emailManager to be responsible for generating oneTimeTokens rather than some coordinating logic class.